tests/api2/pool_dataset_encryption.py

```diff
@@ -0,0 +1,870 @@
+    assert results.status_code == 200, results.text
+
+
+@skip_for_ha
+def test_43_delete_encrypted_dataset():
+    results = DELETE(f'/pool/dataset/id/{dataset_url}/')
+    assert results.status_code == 200, results.text
+
+
+@skip_for_ha
+def test_44_create_an_dataset_with_inherit_encryption_from_the_key_encrypted_pool():
+    payload = {
+        'name': dataset,
+        'inherit_encryption': True
+    }
+    results = POST('/pool/dataset/', payload)
+    assert results.status_code == 200, results.text
+
+
+@skip_for_ha
+def test_45_delete_encrypted_dataset():
+    results = DELETE(f'/pool/dataset/id/{dataset_url}/')
+    assert results.status_code == 200, results.text
+
+
+@skip_for_ha
+def test_46_create_an_encrypted_dataset_with_generate_key_on_key_encrypted_pool():
+    payload = {
+        'name': dataset,
+        'encryption_options': {
+            'generate_key': True,
+        },
+        'encryption': True,
+        'inherit_encryption': False
+    }
+    results = POST('/pool/dataset/', payload)
+    assert results.status_code == 200, results.text
+
+
+@skip_for_ha
+def test_47_delete_generate_key_encrypted_dataset():
+    results = DELETE(f'/pool/dataset/id/{dataset_url}/')
+    assert results.status_code == 200, results.text
+
+
+def test_48_create_a_passphrase_encrypted_root_dataset_parrent():
+    payload = {
+        'name': dataset,
+        'encryption_options': {
+            'passphrase': 'my_passphrase',
+        },
+        'encryption': True,
+        'inherit_encryption': False
+    }
+    results = POST('/pool/dataset/', payload)
+    assert results.status_code == 200, results.text
+
+
+def test_49_create_a_passphrase_encrypted_root_child_of_passphrase_parent():
+    payload = {
+        'name': child_dataset,
+        'encryption_options': {
+            'passphrase': 'my_passphrase2',
+        },
+        'encryption': True,
+        'inherit_encryption': False
+    }
+    results = POST('/pool/dataset/', payload)
+    assert results.status_code == 200, results.text
+
+
+def test_50_lock_passphrase_encrypted_root_with_is_child():
+    payload = {
+        'id': dataset,
+    }
+    results = POST('/pool/dataset/lock', payload)
+    assert results.status_code == 200, results.text
+
+
+def test_51_verify_the_parrent_encrypted_root_is_locked():
+    payload = {
+        'id': dataset
+    }
+    results = POST('/pool/dataset/encryption_summary/', payload)
+    assert results.status_code == 200, results.text
+    job_id = results.json()
+    job_status = wait_on_job(job_id, 120)
+    assert job_status['state'] == 'SUCCESS', str(job_status['results'])
+    job_status_result = job_status['results']['result']
+    for dictionary in job_status_result:
+        if dictionary['name'] == dataset:
+            assert dictionary['locked'] is True, str(job_status_result)
+            break
+    else:
+        assert False, str(job_status_result)
+
+
+def test_52_verify_the_chid_of_the_encrypted_root_parent_is_locked():
```

Comment:
We can verify that the child is locked in the previous test as well using the same data returned